rotkehlchen/accounting/accountant.py

```diff
@@ -228,6 +229,8 @@ def _process_event(
         or with reading the response returned by the server
         """
         ignored_assets = self.db.get_ignored_assets()
+        eth_staking_taxable_after_withdrawal_enabled = db_settings.eth_staking_taxable_after_withdrawal_enabled  # noqa: 501
```

Comment:
No this is not the place to do this. This is a generic event processing function that is called at the very bottom of the processing hierarchy.

The relevant parts you should change is the 2 possible locations where the events can be generated and make them **not taxable** if the setting is on. The 2 places I can think of are:

1. The on-chain eth staking in the beacon chain. The function is here: https://github.com/rotki/rotki/blob/2d2eef424ad16d8b7a7654c04099a7610bf3ad76/rotkehlchen/chain/ethereum/modules/eth2/structures.py#L217-L221 . I guess all you need to do is change the `taxable` 
2. The ETH2 staking in kraken. Here you should check if ETH2 and if setting on and set to `taxable`accordingly: https://github.com/rotki/rotki/blob/2d2eef424ad16d8b7a7654c04099a7610bf3ad76/rotkehlchen/accounting/structures/base.py#L273-L290

Or ... hm ... actually now that I think of it ... you should indeed do this, but perhaps instead of setting it to non taxable you should just omit emitting the acquisition event all together. Since the goal here is to count the ETH2 as acquired **and** taxable from the moment the merge happens.